### PR TITLE
Math Magicians: Added React Tests

### PR DESCRIPTION
**Pull request by Mike111222**
I have done the following changes and I want to implement these:

- [x] Created a new feature branch
- [x] Created separate tests folders in components and logic folders
- [x] Created tests according to the requirements of the project along with Snapshots
- [x] Checked the tests and fixed the errors to make sure all tests are passing
- [x] Fixed Linters Errors
- [x] Updated Calculator.js
- [x] Updated README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,9 +14,9 @@
...
 # 📗 Table of Contents
 
-- [mymathmagiciansapp](#mymathmagiciansapp)
+- [math-magicians](#math-magicians)
 - [📗 Table of Contents](#-table-of-contents)
-- [📖 \[mymathmagiciansapp\] ](#-mymathmagiciansapp-)
+- [📖 \[math-magicians\] ](#-math-magicians-)
   - [🛠 Built With ](#-built-with-)
     - [Tech Stack ](#tech-stack-)
     - [Key Features ](#key-features-)
```
Comment (julie-ify) [inline README.md:22]: - [x] Kindly fix the live demo link in your readme header. It is broken. You can link it to a section or remove it.

File: src/logic/tests/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  test('should return empty obj', () => {
+    expect(calculate(obj, '=')).toEqual({});
+  });
+
+  const obj1 = { total: '16', next: '8', operation: '+' };
+  test('should return obj which total is calculated', () => {
+    expect(calculate(obj1, '=')).toEqual({ total: operate('16', '8', '+'), next: null, operation: null });
+  });
+});
```
Comment (julie-ify) [inline src/logic/tests/calculate.test.js]: - [x] Kindly cover all the scenarios in the calculate.js file

File: src/logic/tests/operate.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    expect(operate(18, 6, '-')).toEqual('12');
+  }); test('multiply', () => {
+    expect(operate(5, 3, 'x')).toEqual('15');
+  }); test('divide', () => {
+    expect(operate(32, 8, '÷')).toEqual('4');
+  });
+  test('mod', () => {
+    expect(operate(21, 5, '%')).toEqual('1');
+  });
+});
```
Comment (julie-ify) [inline src/logic/tests/operate.test.js]: - [x] Kindly cover all the scenarios in the operate.js file

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @yin-ka and @Mike111222,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Mike111222) [reply README.md:22]: I have done the changes you suggested. Thanks a lot for your kind feedback.

File: src/logic/tests/calculate.test.js (hunk 2 above)
Comment (Mike111222) [reply src/logic/tests/calculate.test.js]: I have done the changes you suggested and covered all the scenarios you mentioned. Thanks a lot for your kind feedback.

File: src/logic/tests/operate.test.js (hunk 3 above)
Comment (Mike111222) [reply src/logic/tests/operate.test.js]: I have done the changes you suggested and covered all the scenarios you mentioned. Thanks a lot for your kind feedback.

**Review by Strangeal — APPROVED** [review]
Hi @Mike111222, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Alt Text

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**